packages/client/src/views/SysAdmin/Config/Application/Tabs/GeneralProperties/LoginBackground.tsx

```diff
@@ -0,0 +1,385 @@
+  HalfWidthInput,
+  Label,
+  Message
+} from '@client/views/SysAdmin/Config/Application/Components'
+import { SimpleDocumentUploader } from '@client/components/form/DocumentUploadfield/SimpleDocumentUploader'
+import { GeneralActionId } from '@client/views/SysAdmin/Config/Application'
+import { IAttachmentValue } from '@client/forms'
+
+import { CountryLogo } from '@opencrvs/components/lib/icons'
+import styled from 'styled-components'
+import { InputField } from '@client/components/form/InputField'
+import { Select2 } from '@client/../../components/lib/Select/Select2'
+export enum TabId {
+  COLOUR = 'Colour',
+  IMAGE = 'Image'
+}
+enum BackgroundSize {
+  FILL = 'FILL',
+  TILE = 'TILE'
+}
+const Box = styled.span`
+  width: 40px;
+  height: 40px;
+  margin-left: 5px;
+  border: 2px solid ${({ theme }) => theme.colors.grey300};
+  background-color: ${(p) => (p.color ? p.color : '#fff')};
+`
+const HexInput = styled(HalfWidthInput)`
+  width: 200px;
+`
+export const LoginBackground = () => {
+  const intl = useIntl()
+  const dispatch = useDispatch()
+  const offlineCountryConfiguration = useSelector((store: IStoreState) =>
+    getOfflineData(store)
+  )
+  const [loginBackgroundFileName, setBackgroundFileName] =
+    React.useState(EMPTY_STRING)
+  const [hexValue, setHexValue] = React.useState(EMPTY_STRING)
+  const [isRequestValid, setRequestValid] = React.useState(false)
+  const selectOptions = () => {
+    const arr = [
+      { label: BackgroundSize.FILL, value: BackgroundSize.FILL },
+      { label: BackgroundSize.TILE, value: BackgroundSize.TILE }
+    ]
+    return arr
+  }
+  const [selectedOption, onOptionChange] = React.useState({
+    label: BackgroundSize.FILL,
+    value: BackgroundSize.FILL
+  })
+  const [backgroundType, setBackgroundType] = React.useState(
+    BackgroundSize.FILL
+  )
+  const [showColour, setShowColour] = React.useState(false)
+  const [errorOccured, setErrorOccured] = React.useState(false)
+  const [errorMessages, setErrorMessages] = React.useState(EMPTY_STRING)
+  const [backgroundImage, setBackgroundImage] = React.useState(EMPTY_STRING)
+  const [isFileUploading, setIsFileUploading] = React.useState(false)
+  const [backgroundFile, setBackgroundFile] = React.useState<{
+    name?: string
+    type: string
+    data: string
+  }>({
+    name: EMPTY_STRING,
+    type: EMPTY_STRING,
+    data: EMPTY_STRING
+  })
+  const [notificationStatus, setNotificationStatus] =
+    React.useState<NOTIFICATION_STATUS>(NOTIFICATION_STATUS.IDLE)
+  const [activeTabId, setActiveTabId] = React.useState(TabId.COLOUR)
+
+  const tabSections = [
+    {
+      id: TabId.COLOUR,
+      title: intl.formatMessage(messages.colourTabTitle)
+    },
+    {
+      id: TabId.IMAGE,
+      title: intl.formatMessage(messages.imageTabTitle)
+    }
+  ]
+  const handleColorChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value.toUpperCase()
+
+    if (/^[A-Fa-f0-9]{0,6}$/.test(value)) {
```

Comment:
it could be more useful if could make a util method named something like 'isValidColorCode' and put in '/home/anik/Desktop/opencrvs-core/packages/client/src/views/SysAdmin/Config/Application/utils.ts'

Reply:
I've a question here, why do we need this second if block here?

Reply:
It is used for validation of the input field entry. While the other one is for checking when to show the hex color for 3 or 6 hex value entries.